test/Test/RegistryDAO.hs

```diff
@@ -0,0 +1,150 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.RegistryDAO
+  ( test_RegistryDAO
+  ) where
+
+import Universum hiding (compare, drop, (>>))
+
+import Lorentz
+import Morley.Nettest
+import Morley.Nettest.Tasty
+import Test.Tasty (TestTree, testGroup)
+import Time (sec)
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import Lorentz.Contracts.RegistryDAO
+import Lorentz.Contracts.RegistryDAO.Types
+import Test.Common
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+-- | Testing a registry-like DAO. Ex. DNS registry
+test_RegistryDAO :: TestTree
+test_RegistryDAO = testGroup "RegistryDAO Tests"
+  [ testGroup "Proposal creator:"
+      [ nettestScenario "can propose a valid proposal" validProposal
+      , nettestScenario "can propose a valid configuration proposal" validConfigProposal
+      ]
+    -- TODO: How to test whether or not registry got updated
```

Comment:
Can we test it only on emulator by checking the updated storage?

Reply:
Yes we can do it. We still have to wait for this issue to be able to check the updated storage: https://gitlab.com/morley-framework/morley/-/issues/346

Reply:
So can you add a test or two which would use `eiCheckStorage` from `EmulatedImpl`? It doesn't require the storage to have no big_map.

Reply:
From what I understand `eiCheckStorage` check the whole storage right? Is it possible to simply check a field of the storage, e.g. `sExtra`?

Reply:
Oh, that's stupid actually. I don't see a problem to provide a more general method which would let one check a specific field (e. g. by taking a predicate). Probably it just wasn't needed and nobody bothered to expose it. Adding such a method to `EmulatedImpl` and making it available here will likely take some time. While it's not available, I propose to use the Integrational interface to write a test or two with storage checks. It has more functions to check storage, we can use them. Let's not write too many such tests because we will have to update them to use Nettest at some point.